commands/pull.js

```diff
@@ -0,0 +1,145 @@
+'use strict'
+
+const cli = require('heroku-cli-util')
+const co = require('co')
+const debug = require('debug')('push')
+const psql = require('../lib/psql')
+
+function parseURL (db) {
+  const url = require('url')
+  db = url.parse(db.match(/:\/\//) ? db : `postgres:///${db}`)
+  let [user, password] = db.auth ? db.auth.split(':') : []
+  db.user = user
+  db.password = password
+  db.database = db.path ? db.path.split('/', 2)[1] : null
+  db.host = db.hostname || 'localhost'
+  return db
+}
+
+function exec (cmd, opts = {}) {
+  const {execSync} = require('child_process')
+  debug(cmd)
+  opts = Object.assign({}, opts, {stdio: 'inherit'})
+  try {
+    return execSync(cmd, opts)
+  } catch (err) {
+    if (err.status) process.exit(err.status)
+    throw err
+  }
+}
+
+const prepare = co.wrap(function * (target) {
+  if (target.host === 'localhost') {
+    exec(`createdb ${connstring(target, true)}`)
+  } else {
+    let result = yield psql.exec(target, 'SELECT COUNT(*) = 0 AS empty FROM pg_stat_user_tables')
+    if (!result.startsWith(' empty \n-------\n t')) throw new Error(`Remote database is not empty. Please create a new database or use ${cli.color.cmd('heroku pg:reset')}`)
```

Comment:
yeah this matches existing behavior. It does seem like a strange way to detect if the db is empty.